Symfony/CS/Fixer/Contrib/DirConstantFixer.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Vladimir Reznichenko <kalessil@gmail.com>
+ */
+final class DirConstantFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_FILE);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $end = $tokens->count() - 1;
+
+        $sequenceNeeded = array(array(T_STRING, 'dirname'), '(', array(T_FILE), ')');
+
+        $currIndex = 0;
+        while (null !== $currIndex) {
+            $matches = $tokens->findSequence($sequenceNeeded, $currIndex, $end, false);
+
+            // stop looping if didn't find any new matches
+            if (null === $matches) {
+                break;
+            }
+
+            // 0 to 3 accordingly are "dirname", "(", "__FILE__", ")"
+            $matches = array_keys($matches);
+
+            // move cursor just after sequence
+            $currIndex = $matches[3];
```

Comment:
This is still wrong, as you can have more than 4 tokens in the sequence (because if spaces and comments)